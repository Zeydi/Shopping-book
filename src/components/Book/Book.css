.info {
  font-size: 15px;
  padding: 5px;
  text-align: justify;
}
.card-title {
  height: 120px;
}
.link {
  margin-top: 40px;
  cursor: pointer;
}
