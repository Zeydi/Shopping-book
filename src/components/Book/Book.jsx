import React, { Component } from "react";
import "./Book.css";

class Book extends Component {
  constructor(props) {
    super(props);
    this.state = { isOpen: false };
  }

  toggleReveal = () => {
    this.setState(prevState => ({ isOpen: !prevState.isOpen }));
  };

  renderResume = () => {
    const { title, info } = this.props;

    return (
      <div>
        <h6 className="grey-text text-darken-4">
          {title}
          <i
            className="material-icons   link"
            onClick={() => this.toggleReveal()}
          >
            close
          </i>
        </h6>
        <p className="info">{info}</p>
      </div>
    );
  };

  render() {
    const { isOpen } = this.state;
    const { title, cover, price, id } = this.props;
    return (
      <div className="card cover">
        {!isOpen ? (
          <div>
            <div className="card-image">
              <img src={cover} alt={title} />
            </div>
            <div className="card-content">
              <span className="card-title ">{title}</span>
              <p>
                <b>Price: {price}$</b>
              </p>
              <div className="card-action link">
                <span id="toggle" onClick={() => this.toggleReveal()}>
                  Show resume
                </span>
              </div>
              <span
                to="/"
                className="btn-floating halfway-fab waves-effect waves-light red"
                onClick={() => {
                  this.props.handleClick(id);
                }}
              >
                <i className="material-icons">add</i>
              </span>
            </div>
          </div>
        ) : (
          this.renderResume()
        )}
      </div>
    );
  }
}

export default Book;
