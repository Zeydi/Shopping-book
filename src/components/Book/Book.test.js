import React from "react";
import { shallow } from "enzyme";
import Book from "./Book";

describe("Book component", () => {
  const mockHandleClick = jest.fn();

  const book = shallow(<Book handleClick={mockHandleClick} />);

  it("renders correctly", () => {
    expect(book).toMatchSnapshot();
  });

  it("button click should call function passed as props", () => {
    book.find(".btn-floating").simulate("click");
    expect(mockHandleClick).toHaveBeenCalled();
  });

  it("should toggle isopen when clicking", () => {
    expect(book.state().isOpen).toBe(false);

    const toggle = book.find("#toggle");
    toggle.simulate("click");
    expect(book.state().isOpen).toBe(true);

    toggle.simulate("click");
    expect(book.state("isOpen")).toEqual(false);
  });
});
