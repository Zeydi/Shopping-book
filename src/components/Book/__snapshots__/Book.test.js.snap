// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book component renders correctly 1`] = `
<div
  className="card cover"
>
  <div>
    <div
      className="card-image"
    >
      <img />
    </div>
    <div
      className="card-content"
    >
      <span
        className="card-title "
      />
      <p>
        <b>
          Price: 
          $
        </b>
      </p>
      <div
        className="card-action link"
      >
        <span
          id="toggle"
          onClick={[Function]}
        >
          Show resume
        </span>
      </div>
      <span
        className="btn-floating halfway-fab waves-effect waves-light red"
        onClick={[Function]}
        to="/"
      >
        <i
          className="material-icons"
        >
          add
        </i>
      </span>
    </div>
  </div>
</div>
`;
